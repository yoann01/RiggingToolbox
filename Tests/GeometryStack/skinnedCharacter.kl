
require RiggingToolbox;

operator entry(){

  StartFabricProfiling();

  GeometryStack stack();

  AlembicSkinnedMeshGeometryGenerator alembicGenerator();
  SkinningModifier skiningModifier();
  stack.addGeometryOperator(alembicGenerator);
  stack.addGeometryOperator(skiningModifier);

  alembicGenerator.setFilePath("${FABRIC_RIGGINGTOOLBOX_PATH}/Tests/GeometryStack/Resources/skinnedTube.abc");

  String geometryNames[];
  geometryNames.push("pCylinderShape1");
  alembicGenerator.setGeometryNames(geometryNames);

  EvalContext context();
  GeometrySet geomSet = stack.evaluate(context);

  // Modify the pose and then reevaluate. 
  Mat44 pose[];
  pose.resize(2);
  pose[0] = Xfo(Vec3(3, 4, 5)).toMat44();
  pose[1] = Xfo(Vec3(10, 20, 5)).toMat44();
  skiningModifier.setPose(pose);

  stack.evaluate(context);

  StopFabricProfiling();

  report( GetEvalPathReport() );
}

