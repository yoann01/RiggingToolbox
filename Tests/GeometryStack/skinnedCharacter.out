Importing:D:/Projects/FabricEngineInc/RiggingToolbox/Tests/GeometryStack/Resources/skinnedTube.abc
Warning: Pose count does not match the reference pose count. referencePose:4 != pose:2. Skinning disabled
function GeometryStack.notify!(Notifier, String, String):AlembicSkinnedMeshGeometryGenerator.changed
function GeometryStack.notify!(Notifier, String, String):AlembicSkinnedMeshGeometryGenerator.changed
function GeometrySet GeometryStack.evaluate!(EvalContext)
--function GeometryCache.update!(io GeometrySet, GeometryOperator)
--function AlembicSkinnedMeshGeometryGenerator.evaluate!(EvalContext, io GeometrySet)
--function GeometryAttributeCache.update!(io GeometrySet, GeometryOperator):["positions"]
----Update:positions
--function SkinningModifier.evaluate!(EvalContext, io GeometrySet)
function GeometryStack.notify!(Notifier, String, String):SkinningModifier.changed
function GeometrySet GeometryStack.evaluate!(EvalContext)
--function GeometryAttributeCache.update!(io GeometrySet, GeometryOperator):["positions"]
----Restore:positions
--function SkinningModifier.evaluate!(EvalContext, io GeometrySet)

